File: FreelanceRestApi/src/main/java/com/hamzabekkaoui/freelancerestapi/dto/request/FreelanceRegistrationRequest.java
package com.hamzabekkaoui.freelancerestapi.dto.request;

import lombok.Builder;

import java.util.List;

@Builder
public record FreelanceRegistrationRequest(
        String firstName,
        String lastName,
        String userName,
        String email,
        String phoneNumber,
        String city,
        String jobTitle,
        String imageUrl,
        String cvUrl,
        List<TechnologyRequest> technologies,
        String password,
        String confirmationPassword
) {
}

File: FreelanceFrontEnd/src/app/types/FreelanceRegistration-request.ts
import {TechnologyRequest} from "./Technology-request";

export interface FreelanceRegistrationRequest{
  id?:number,
  firstName:string,
  lastName:string,
  userName:string,
  city:string,
  jobTitle:string,
  email:string,
  phoneNumber:string,
  imageUrl?:string,
  cvUrl?:string,
  technologyRequest?:TechnologyRequest[]
  password:string,
  confirmationPassword:string
}



